### DISCUSSION TITLE
Unable to read file from a location in the variable value in XML body

### DISCUSSION THREAD

**Discussion by Sateswar** [thunderclient/thunder-client-support#1023] [Q&A]
![image](https://user-images.githubusercontent.com/7892396/220466434-23bd2fb8-50bc-4be9-8fbb-8d76e11a513b.png)
![image](https://user-images.githubusercontent.com/7892396/220465946-8e1dcfdc-2549-4b10-ac0c-36101353e634.png)
![image](https://user-images.githubusercontent.com/7892396/220466244-595e6b25-f769-443c-b6f7-95b3858c9584.png)

Please let me know if I'm wrong in updating the path in variable

**Comment by rangav** [comment]
Hi @Sateswar looks like there is extra line break at the end of file path value, please remove any extra spaces in the variable value

![220466244-595e6b25-f769-443c-b6f7-95b3858c9584](https://user-images.githubusercontent.com/8637550/220468453-b3567589-7fe9-4440-88a9-ecfcf7faf7ce.png)

**Comment by rangav** [comment]
also `remove the quotes` in the value

**Reply by Sateswar** [reply]
Thank you so much! 

When I deleted /n it gave me another issue with API url variable. Then, uninstalled and deleted the program data then installed the plugin. After these steps it started working.

Looks like the issues was in configuration.

**Reply by Sateswar** [reply]
Sorry to reopen this. Actually I had problem with reading .xml files, after your comment I verified .json file which worked fine. However, with you recommendation when I retried .xml file, I'm seeing a new error. Screenshot attached below.

I believe it is complaining about the API url, however, I double checked the API url and it is correct.

![image](https://user-images.githubusercontent.com/7892396/221094677-4eda809a-ebdd-4865-88c7-e913fc5c9307.png)
![image](https://user-images.githubusercontent.com/7892396/221094839-3e53b00c-3587-4f2f-8af1-89c6d203c82e.png)
![image](https://user-images.githubusercontent.com/7892396/221094948-5614969c-a511-42c0-a1ef-51f26169c9cb.png)

**Reply by rangav** [reply]
Readfile only works for local files, so download the file to your project and use that path.

**Reply by Sateswar** [reply]
I think you misunderstood.  GetTranslatedDataUsingXmlPath is the API URL not the variable for file path.

**Reply by rangav** [reply]
oh ok sorry,

Can you share your request screenshot? Will look into it.

**Reply by rangav** [reply]
few more questions?
- which env this variable is stored?
- Is it current active environment?
